src/port1.0/portconfigure.tcl

```diff
@@ -1552,6 +1554,15 @@ proc portconfigure::add_automatic_compiler_dependencies {} {
         ui_debug "Adding Fortran compiler dependency"
         portconfigure::add_compiler_port_dependencies [portconfigure::configure_get_fortran_compiler]
     }
+
+    # Clang (LLVM or Xcode) requires OpenMP runtime from libomp
+    # (GCC will use OpenMP runtime from libgcc)
+    if {[option compiler.openmp_version] ne "" && [string match *clang* ${configure.compiler}]} {
+        ui_debug "Adding depends_lib port:libomp"
+        depends_lib-delete "port:libomp"
+        depends_lib-append "port:libomp"
+    }
```

Comment:
The automatic libomp dependency currently only applies to MacPorts clang, so I have tried relocating it to also apply to Xcode clang, but I am not certain I have done so correctly.